benchmarks/definition/config.py

```diff
@@ -88,3 +89,17 @@ def __call__(self) -> Result:
         )
 
         return Result(self.name, result, metadata)
+
+    @staticmethod
+    def serialize(benchmark: "Benchmark") -> dict:
```

Comment:
Not sure about this, but do we actually require the inheritance from `SerialMixin` if we have a custom serialization method?

Reply:
I've added a few changes to the way the classes are serialized.

Reply:
To be honest, I liked it better previously - now there are two different serialization mechanisms within BayBE, one in the benchmarks and one here. This is just confusing and should be avoided in my opinion. Also, my comment was more meant as a question and not as a "please do that". If I want changes, I explicitly say so :D
So in this case, I would prefer to go back to the previous version. Can we maybe just override the corresponding `to_json`, `to_dict` functions? Would however be interested in @AdrianSosic opinion on this, since maybe having a dedicated second serialization mechanism actually is the preferable option.

Reply:
That change was suggested by @AdrianSosic, if I understood him correctly. Sorry, I forgot to add that detail to the comment.